packages/ui/src/models/visualization/flows/abstract-camel-visual-entity.ts

```diff
@@ -72,7 +72,8 @@ export abstract class AbstractCamelVisualEntity implements BaseVisualCamelEntity
     const updatedValue = CamelComponentSchemaService.getUriSerializedDefinition(path, value);
 
     setValue(this.route, path, updatedValue);
-    this.id = this.route.id;
+
+    if (isDefined(this.route.id)) this.id = this.route.id;
```

Comment:
@lordrip 
Can we write this as follows as updateModel() ensure that form text input can never be undefined.
this.id = this.route.id!;